rpcserver.go

```diff
@@ -1686,6 +1721,49 @@ func (r *rpcServer) OpenChannel(in *lnrpc.OpenChannelRequest,
 			if err != nil {
 				return err
 			}
+
+		// If we have a PSBT shim, then this means the funding
+		// transaction will be crafted outside of the wallet, once the
+		// funding multisig output script is known. We'll create an
+		// intent that will supervise the multi-step process.
+		case in.FundingShim.GetPsbtShim() != nil:
+			psbtShim := in.FundingShim.GetPsbtShim()
+
+			// Instruct the wallet to use the new
+			// chanfunding.PsbtAssembler to construct the funding
+			// transaction.
+			psbtAssembler, err := newPsbtAssembler(psbtShim)
+			if err != nil {
+				return err
+			}
+			req.chanFunder = psbtAssembler
+
+			// We'll also need to inform the wallet about the
+			// funding intent, so ask the assembler to provision it
+			// for us.
+			psbtIntent, err := psbtAssembler.ProvisionChannel(
```

Comment:
Does this need to be provisioned _outside_ the wallet? Or is it that the later steps depend on the wallet already having stored the intent in the internal map for retrieval purposes?

Reply:
Ah, no, it's not needed. This was left in from my misunderstanding of how `GetShimRegister` was used.